llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -2595,6 +2595,10 @@ X86TargetLowering::X86TargetLowering(const X86TargetMachine &TM,
         setOperationAction(Op, MVT::f32, Promote);
   }
 
+  setOperationPromotedToType(ISD::ATOMIC_LOAD, MVT::f16, MVT::i16);
```

Comment:
Presumably similar changes to other backends are also required?

Reply:
Handle bf16 as well?

Reply:
`bf16` is already lowered properly without promotion.

And yes, other backends would either have to promote these here or implement them explicitly.

Reply:
SelectionDAG making anything legal by default was a terrible mistake but we're not going to fix that here